gsd/test/test_fl.py

```diff
@@ -961,3 +961,61 @@ def test_read_write(tmp_path, mode):
             data10 = f.read_chunk(frame=i, name='data10')
             assert data1[0] == i
             assert data10[0] == i * 10
+
+
+@pytest.mark.parametrize('n_flush', [0, 1, 2])
+def test_flush(tmp_path, open_mode, n_flush):
+    """Test flush."""
+    data = numpy.array([1, 2, 3, 4, 5, 10012], dtype=numpy.int64)
+    with gsd.fl.open(name=tmp_path / 'test_flush.gsd',
+                     mode=open_mode.write,
+                     application='test_flush',
+                     schema='none',
+                     schema_version=[1, 2]) as f:
+        f.write_chunk(name='chunk1', data=data)
+        f.end_frame()
+        f.write_chunk(name='chunk2', data=data)
+        f.end_frame()
+        f.write_chunk(name='chunk3', data=data)
+
+        # 0 calls to flush tests the implicit flush on close, 2 calls to flush
+        # tests that repeated calls are handled properly.
+        for i in range(n_flush):
+            f.flush()
+
+    with gsd.fl.open(name=tmp_path / 'test_flush.gsd',
+                     mode=open_mode.read) as f:
+        assert f.chunk_exists(frame=0, name='chunk1')
+        assert f.chunk_exists(frame=1, name='chunk2')
+
+        # The third chunk is not written because end_frame is not called a
+        # third time.
+
+        assert not f.chunk_exists(frame=2, name='chunk3')
+        assert f.nframes == 2
+
+
+def test_maximum_write_buffer_size(tmp_path, open_mode):
```

Comment:
Let's test that setting to values `==0` errors. Same below.

Reply:
Done.